atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java

```diff
@@ -106,6 +109,11 @@ protected int getPoolSize() {
             if (resource != null) {
                 if (shouldReuse) {
                     log.debug("Returning {} to pool", resource);
+                    TTransport transport = resource.getInputProtocol().getTransport();
+                    if (transport instanceof TMemoryInputTransport) {
+                        log.debug("During {} check-in, cleaned up an iprot of size {}", resource, transport.getBuffer().length);
```

Comment:
TMemInputTransport is the one that the buffers-that-stay-around-after-reads appear as; unlike reads, I assume it's better at self-policing it's own write buffers, but I can look into it.

I doubt the buffer will be null, but no sense not null-checking it, you're right.